pkg/redis/RedisConnectionFactory.php

```diff
@@ -143,4 +144,23 @@ private function defaultConfig()
             'database' => 0,
         ];
     }
+
+    /**
+     * @param $redis
+     */
+    public function setRedis($redis)
```

Comment:
I am not in favor of a setter. The RedisConnectionFactory constructor should handle all this.